plugins/liquid/refined-types-plugin/src/main/kotlin/arrow/meta/plugins/liquid/phases/analysis/solver/collect/ConstraintCollection.kt

```diff
@@ -217,7 +329,10 @@ internal fun KtDeclaration.constraints(
 private fun KtElement.constraintsDSLElements(): Set<PsiElement> {
   val results = hashSetOf<PsiElement>()
   val visitor = callExpressionRecursiveVisitor {
-    if (it.calleeExpression?.text == "pre" || it.calleeExpression?.text == "post") {
+    if (it.calleeExpression?.text == "pre" ||
+      it.calleeExpression?.text == "post" ||
+      it.calleeExpression?.text == "require"
```

Comment:
this partially addresses https://app.clickup.com/t/p5eyun at least the `require` part. The other part is related to contracts; we can perhaps create a separate issue and look into it after our first release.

Reply:
Indeed. We might want to separate this in two different tasks.